src/cljam/io/bam_index/writer.clj

```diff
@@ -76,11 +77,11 @@
 (defn- update-bin-index
   [bin-index aln]
   (let [bin (sam-util/compute-bin aln)
-        achunk (:chunk (:meta aln))]
+        achunk ^Chunk (chunk/map->Chunk (:chunk (:meta aln)))]
```

Comment:
Record's `map->Foo` function is slow. I recommend using `->Foo` function or Java constructor `Foo.`.